toolchain/check/import_ref.cpp

```diff
@@ -694,24 +697,30 @@ class ImportRefResolver {
     // incomplete type so that any references have something to point at.
     class_decl.class_id = context_.classes().Add({
         .name_id = GetLocalNameId(import_class.name_id),
-        // Set in the second pass once we've imported them.
+        // These are set in the second pass once we've imported them. Import
+        // enough of the parameter lists that we know whether this class is a
+        // generic class and can build the right constant value for it.
         .enclosing_scope_id = SemIR::NameScopeId::Invalid,
-        .implicit_param_refs_id = SemIR::InstBlockId::Invalid,
-        .param_refs_id = SemIR::InstBlockId::Invalid,
-        // `.self_type_id` depends on the ClassType, so is set below.
+        .implicit_param_refs_id = import_class.implicit_param_refs_id.is_valid()
+                                      ? SemIR::InstBlockId::Empty
+                                      : SemIR::InstBlockId::Invalid,
+        .param_refs_id = import_class.param_refs_id.is_valid()
+                             ? SemIR::InstBlockId::Empty
+                             : SemIR::InstBlockId::Invalid,
```

Comment:
Setting Empty as incorrect-but-fully-functional leaves me concerned that it exposes things to incorrect reads. Had you considered a different special value, such as a new Uninitialized (incorrect-and-broken), or perhaps expanding ClassDecl to have a GenericClassDecl (assuming the below ReplaceInstBeforeConstantUse is at the root of this)?